src/sidebar/components/version-info.js

```diff
@@ -39,7 +39,7 @@ function VersionInfo({ flash, versionData }) {
       </dl>
       <div className="version-info__actions">
         <button className="version-info__copy-btn" onClick={copyVersionData}>
-          <SvgIcon name="copy" className="version-info__copy-btn-icon" />
+          <SvgIcon name="copy" className="version-info__copy-btn__icon" />
```

Comment:
hmm, I'm a bit confused by this, I thought we only used one double underscore. Also, if this class is not referenced in any css file, can we omit it entirely?

Reply:
Yeah, this is a tricky one. A `VersionInfo` component has a `copy-btn`, so you'd expect a classname like `version-info__copy-btn`. But the `copy-btn` subsequently has an `icon`— that `icon` belongs to the `copy-btn`, not to `version-info`. So then you have a head-scratcher like `version-info__copy-btn__icon`.

In `VersionInfo`'s SCSS, the rule for `version-info__copy-btn` `@include`s the `action-button` mixin. Inside of _that_ mixin is a rule for `&__icon`. So we're kind of nested here. It makes me feel like I'm organizing this wrong, but I don't know what to do in our current convention landscape. It may be time for `action-button` to be its own reusable sub-component (style-wise) akin to `primary-action-button`, but the way `primary-action-button`'s SCSS is treated (like it's for a component, but there's no actual corresponding JS component) is unique and I'm not sure it's a pattern we want to repeat? Or perhaps adjust? Put another way, I find it a little confusing to have SCSS modules in the `components` directory that don't have a corresponding JS component; I wonder if we could put these "components" (composite, reusable SCSS mixins made into utility-like reusable classes) somewhere else? I think I'm getting a little tangled up and may be overthinking this.

Reply:
awe, the mixin, ya thats hard. But I think we're on the right track, maybe its just not fully flushed out. I do think that we should be created empty bare bone js counter parts for this stuff that we can build on top of or override parts of. Lets leave this for now and we can improve as we migrate into a formal pattern lib

Reply:
> but the way primary-action-button's SCSS is treated (like it's for a component, but there's no actual corresponding JS component) is unique and I'm not sure it's a pattern we want to repeat?

I think it is totally fine to have a CSS-only component if the markup is trivial (eg. just a plain element). We have such things in both the lms app and h. That said, encapsulating details in a Preact component is very cheap to do, so I suggest we do that quite eagerly.